File: android/PlayKit/app/src/main/java/io/streamroot/dna/playkit/PlayKitBandwidthListener.java
package io.streamroot.dna.playkit;

import com.kaltura.playkit.Player;
import io.streamroot.dna.core.BandwidthListener;

public final class PlayKitBandwidthListener implements BandwidthListener {
    private Player mPlayer;

    public PlayKitBandwidthListener(Player player) {
        this.mPlayer = player;
    }

    @Override
    public void onBandwidthChange(long estimatedBandwidth) {
        this.mPlayer.setEstimatedBandwidth(estimatedBandwidth);
    }
}

File: android/PlayKit/app/src/main/java/io/streamroot/dna/playkit/PlaykitBandwidthListener.kt
package io.streamroot.dna.playkit

import com.kaltura.playkit.Player
import io.streamroot.dna.core.BandwidthListener

class PlaykitBandwidthListener(
    player: Player
): BandwidthListener {
    private val mPlayer = player

    override fun onBandwidthChange(estimatedBandwidth: Long) {
        mPlayer.estimatedBandwidth = estimatedBandwidth
    }
}